pkg/kubelet/config/file_linux.go

```diff
@@ -40,7 +41,15 @@ const (
 	podDelete
 )
 
-func (s *sourceFile) watch() error {
+func (s *sourceFileListerWatcher) watch() {
+	go wait.Forever(func() {
+		if err := s.doWatch(); err != nil {
+			glog.Errorf("unable to read config path %q: %v", s.path, err)
+		}
+	}, s.period)
```

Comment:
`s.period` could be replaced by a static duration.
A loop around `doWatch` is to keep it continue and retry after err happened like the watched path doesn't exist at the beginning.